api/async.rb

```diff
@@ -103,4 +107,40 @@ def validate
       end
     end
   end
+
+  # Many resources are async but do not provide a long running Operation that can be
+  # polled against. This class is meant to poll the resource itself until it can be
+  # determined that the resource is in a ready state.
+  class PollAsync < Async
```

Comment:
Can you note that an operation handler with a certain name + signature is expected to be handwritten?